runtime/manifest-parser.peg

```diff
@@ -670,34 +692,67 @@ Direction
   = dir:('<-' / '->' / '=')
 
 ConnectionTarget
-  = ident:(upperIdent / lowerIdent) param:('.' lowerIdent)? tags:(whiteSpace TagList)?
+  = VerbConnectionTarget / TagConnectionTarget / NameConnectionTarget / ParticleConnectionTarget
+
+VerbConnectionTarget
+  = "can" whiteSpace verbs:TagList components:ConnectionTargetHandleComponents
   {
-    let particle = null;
-    let name = null;
-    if (ident) {
-      if (/^[A-Z]/.test(ident)) {
-        particle = ident;
-      } else {
-        name = ident;
-      }
+    let {param, tags} = components;
+    return {
+      kind: 'connection-target',
+      location: location(),      
+      targetType: 'verb',
+      verbs,
+      param,
+      tags
+    }
+  }
+
+TagConnectionTarget
+  = "is" whiteSpace tags:TagList {
+    return {
+      kind: 'connection-target',
+      location: location(),      
+      targetType: 'tag',
+      tags
     }
+  }
 
+NameConnectionTarget
+  = name:lowerIdent components:ConnectionTargetHandleComponents
+  {
+    let {param, tags} = components;
     return {
       kind: 'connection-target',
+      targetType: 'localName',
       location: location(),
-      particle,
       name,
-      param: optional(param, param => param[1], null),
-      tags: optional(tags, tags => tags[1], []),
+      param,
+      tags
     }
   }
-  / tags:TagList
+
+ParticleConnectionTarget
+  = particle:upperIdent components:ConnectionTargetHandleComponents 
   {
+    let {param, tags} = components;    
     return {
       kind: 'connection-target',
+      targetType: 'particle',
       location: location(),
-      tags,
-    };
+      particle,
+      param,
+      tags
+    }
+  }
+
+ConnectionTargetHandleComponents
+  = param:('.' lowerIdent)? tags:(whiteSpace TagList)?
```

Comment:
Should `.` be optional here? I think we won't parse `can #foo.#bar` yet, but will parse 'name #tag' (which I don't think we should).

```
ConnectionTargetHandleComponents
  = '.' lowerIdent tags:(whiteSpace TagList)?
  { .. }
  / '.' tags:TagList
  { .. }
```